src/main/java/org/bytedeco/javacpp/tools/Parser.java

```diff
@@ -2440,32 +2444,8 @@ boolean function(Context context, DeclarationList declList) throws ParserExcepti
                 }
             }
 
-            // check for const, other attributes, and pure virtual functions, ignoring the body if present
-            for (Token token = tokens.get(); !token.match(Token.EOF); token = tokens.get()) {
-                if (token.match(Token.CONST, Token.__CONST, Token.CONSTEXPR)) {
-                    decl.constMember = true;
-                    token = tokens.next();
-                } else if (token.match(Token.OVERRIDE)) {
-                    type.virtual = true;
-                    // token = tokens.next();
-                    // let through for user defined annotations
-                }
-                if (token.match('&', "&&")) {
-                    // ignore?
-                    token = tokens.next();
-                }
-                Attribute attr = attribute();
-                if (attr != null && attr.annotation) {
-                    dcl.type.annotations += attr.javaName;
-                } else if (attr == null) {
-                    break;
-                }
-            }
-            if (tokens.get().match("->")) {
-                // auto type
-                tokens.next();
-                type = type(context);
-            }
+            type = postDeclarator(context, decl, dcl, type);
```

Comment:
And if the first call is enough, why do we need a second call here?

Reply:
`decl`, `dcl` and `type` are reset in the `for (int n...` loop. The whole declaration is reparsed at each iteration. 
 
And we need to swallow this part of the declaration before continuing the parsing.

Reply:
Hum, what happens if you move that in declarator(), just before `return dcl`, and only call it only when dcl.parameters is not null?

Reply:
`declarator` is called a lot of times, not only for functions, and this post-declarator part is function-specific.

Reply:
Could `declarator` be called from somewhere else than `function` and this post-declarator part is expected ?

Reply:
Probably not, but there are other function-only related things in there because it gets called in a loop in function(). The const part, for example, gets picked up there too.

Reply:
But we need that information before calling parameters() so I think that's not going to help anyways.

Reply:
Right. 
And the post-declarator set `constMember` which is in `decl`, not `dcl`.
Wha is the problem with putting this part of parsing in postDeclarator ?